src/lib/draw-and-pick.js

```diff
@@ -87,27 +87,53 @@ export function pickLayers(gl, {
     });
 
     // Read color in the central pixel, to be mapped with picking colors
-    const color = new Uint8Array(4);
-    gl.readPixels(deviceX, deviceY, 1, 1, GL.RGBA, GL.UNSIGNED_BYTE, color);
+    const pickedColor = new Uint8Array(4);
+    gl.readPixels(deviceX, deviceY, 1, 1, GL.RGBA, GL.UNSIGNED_BYTE, pickedColor);
 
     // restore blend mode
     setBlendMode(gl, oldBlendMode);
     // Picking process end
 
     // Process picked info start
     // Decode alpha to layer index
-    const pickedLayer = layers[color[3] - 1];
+    const pickedLayer = pickedColor[3] ? layers[pickedColor[3] - 1] : null;
+    const affectedLayers = [pickedLayer];
```

Comment:
What if pickedLayer is null? I guess we are filtering below. Since we only have one or two layers now, I think it could be easier to read this code if we explicitly only add the layer here if it is non null.